templates/accounts.go

```diff
@@ -26,67 +26,118 @@ import (
 )
 
 const createAccountTemplate = `
-transaction(publicKeys: [[UInt8]], code: [UInt8]) {
-  prepare(signer: AuthAccount) {
-	let acct = AuthAccount(payer: signer)
+transaction(publicKeys: [[UInt8]]) {
+	prepare(signer: AuthAccount) {
+		let acct = AuthAccount(payer: signer)
 
-	for key in publicKeys {
-	  acct.addPublicKey(key)
+		for key in publicKeys {
+			acct.addPublicKey(key)
+		}
 	}
-	
-	if code.length > 0 {
-	  acct.setCode(code)
+}
+`
+
+const createAccountWitContractsTemplate = `
+transaction(publicKeys: [[UInt8]], contracts: {String: [UInt8]}) {
+	prepare(signer: AuthAccount) {
+		let acct = AuthAccount(payer: signer)
+
+		for key in publicKeys {
+			acct.addPublicKey(key)
+		}
+
+		for contract in contracts.keys {
+			acct.contracts.add(name: contract, code: contracts[contract]!)
+		}
 	}
-  }
 }
 `
 
 // CreateAccount generates a transactions that creates a new account.
 //
-// This template accepts a list of public keys and a code argument, both of which are optional.
+// This template accepts a list of public keys and a contracts argument, both of which are optional.
+//
+// The contracts argument is a dictionary of *contract name*: *contract code (in bytes)*.
+// All of the contracts will be deployed to the account.
 //
 // The final argument is the address of the account that will pay the account creation fee.
 // This account is added as a transaction authorizer and therefore must sign the resulting transaction.
-func CreateAccount(accountKeys []*flow.AccountKey, code []byte, payer flow.Address) *flow.Transaction {
+func CreateAccount(accountKeys []*flow.AccountKey, contracts []Contract, payer flow.Address) *flow.Transaction {
 	publicKeys := make([]cadence.Value, len(accountKeys))
 
 	for i, accountKey := range accountKeys {
 		publicKeys[i] = bytesToCadenceArray(accountKey.Encode())
 	}
 
+	contractKeyPairs := make([]cadence.KeyValuePair, len(contracts))
+
+	for i, contract := range contracts {
+		contractKeyPairs[i] = cadence.KeyValuePair{
+			Key:   cadence.NewString(contract.Name),
+			Value: bytesToCadenceArray(contract.Source),
+		}
+	}
+
 	cadencePublicKeys := cadence.NewArray(publicKeys)
-	cadenceCode := bytesToCadenceArray(code)
+	cadenceContracts := cadence.NewDictionary(contractKeyPairs)
 
 	return flow.NewTransaction().
-		SetScript([]byte(createAccountTemplate)).
+		SetScript([]byte(createAccountWitContractsTemplate)).
 		AddAuthorizer(payer).
 		AddRawArgument(jsoncdc.MustEncode(cadencePublicKeys)).
-		AddRawArgument(jsoncdc.MustEncode(cadenceCode))
+		AddRawArgument(jsoncdc.MustEncode(cadenceContracts))
+}
+
+type Contract struct {
```

Comment:
Can you add a comment to this? 

Something like:

```
// Contract is a Cadence contract deployed to a Flow account.
//
// Name is the identifier for the contract within the account.
//
// Source is the UTF-8 encoded source code for the contract.
```

I would also define this struct before the `CreateAccount` function

Reply:
@turbolent @janezpodhostnik Do you think that `Source` should just be a string? We originally chose bytes so that it'd be easier to swap this out for compiled code (e.g. WASM) in the future, but I'm wondering if this specific struct can just accept a string for convenience. 

The name `Source` does imply that it is source code, which to me also implies that it is a human readable string. Perhaps there could be a separate `Compiled` field in the future.

Reply:
I agree Source would be easier to use if it was a string. I also added a GetSourceBytes function to the Contract for ease of use.

Reply:
It's bytes in the protobuf and state, but I think it's convenient that it's a high-level string in the SDK.